pkg/controllers/multiclusterservice/endpointslice_dispatch_controller.go

```diff
@@ -169,9 +170,71 @@ func (c *EndpointsliceDispatchController) SetupWithManager(mgr controllerruntime
 	}
 	return controllerruntime.NewControllerManagedBy(mgr).For(&workv1alpha1.Work{}, builder.WithPredicates(workPredicateFun)).
 		Watches(&networkingv1alpha1.MultiClusterService{}, handler.EnqueueRequestsFromMapFunc(c.newMultiClusterServiceFunc())).
+		Watches(&clusterv1alpha1.Cluster{}, handler.EnqueueRequestsFromMapFunc(c.newClusterFunc())).
 		Complete(c)
 }
 
+func (c *EndpointsliceDispatchController) newClusterFunc() handler.MapFunc {
+	return func(ctx context.Context, a client.Object) []reconcile.Request {
+		var clusterName string
+		switch t := a.(type) {
+		case *clusterv1alpha1.Cluster:
+			clusterName = t.Name
+		default:
+			return nil
+		}
+
+		mcsList := &networkingv1alpha1.MultiClusterServiceList{}
+		if err := c.Client.List(context.TODO(), mcsList, &client.ListOptions{}); err != nil {
+			klog.Errorf("Failed to list MultiClusterService, error: %v", err)
+			return nil
+		}
+
+		var requests []reconcile.Request
+		for _, mcs := range mcsList.Items {
+			clusterSet, err := helper.GetConsumptionClustres(c.Client, mcs.DeepCopy())
+			if err != nil {
+				klog.Errorf("Failed to get provision clusters, error: %v", err)
+				continue
+			}
+
+			if !clusterSet.Has(clusterName) {
+				continue
+			}
+
+			workList, err := c.getClusterEndpointSliceWorks(mcs.Namespace, mcs.Name)
+			if err != nil {
+				klog.Errorf("Failed to list work, error: %v", err)
+				continue
+			}
+			for _, work := range workList {
+				// We only care about the EndpointSlice work from provision clusters
+				if util.GetLabelValue(work.Annotations, util.EndpointSliceProvisionClusterAnnotation) != "" {
```

Comment:
Seems the comments mismatch with the action.
A work with `endpointslice.karmada.io/provision-cluster` means it is collected from the provision cluster, but why it is ignored?

Reply:
this annotation is added to the dispatch work(in consumption clusters).
For the work in provision clusters, this annotation will not be added. 
I can add more comment to explain more details

Reply:
Or we can add other annotation, like:
for the EndpointSlice work in provision cluster execution namespace, we add annotation: `endpointslice.karmada.io/role: original`
for the dispatched EndpointSlice work in consumption cluster execution namespace, we add annotation: `endpointslice.karmada.io/role: dispatched`